src/main/java/iotbay/controller/SearchProductsController.java

```diff
@@ -1,20 +1,56 @@
 package iotbay.controller;
 
+import iotbay.helper.ProjectConstants;
+import iotbay.model.Product;
+import iotbay.service.ProductService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.sql.Connection;
+import java.util.List;
+import java.util.stream.Collectors;
 
-// Controller class for the searchProducts.jsp page
+@WebServlet("/SearchProducts")
 public class SearchProductsController extends HttpServlet {
-    // Using the GET method for searching products
-    public void doGet(HttpServletRequest request, HttpServletResponse response) {
-        // Fetch session object from request
-        // Fetch connection object from session
-        // Create an instance of the ProductService class
-        // get search term from request object
-        // call .getAllProducts() method on ProductService object
-        // Filter the product list for products whose name contains the search term
-        // Store productList in session
-        // redirect to searchProducts.jsp page
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        // Get current session
+        HttpSession session = request.getSession();
+
+        // Retrieve DB connection object from session using constant key
+        Connection connection = (Connection) session.getAttribute(ProjectConstants.SESSION_ATTRIBUTE_DBCONNECTION);
+
+        if (connection == null) {
+            // If connection missing, redirect (could be login or error page)
+            response.sendRedirect("login.jsp");
```

Comment:
This doesn't change anything functionally but can you use the ProjectConstants thing I made here
```suggestion
            response.sendRedirect(ProjectConstants.LOGIN_PAGE);
```